fdbserver/ClusterRecovery.actor.cpp

```diff
@@ -1080,8 +1079,17 @@ ACTOR Future<Void> readTransactionSystemState(Reference<ClusterRecoveryData> sel
 	// Sets self->configuration to the configuration (FF/conf/ keys) at self->lastEpochEnd
 
 	// Recover transaction state store
-	bool enableEncryptionForTxnStateStore = isEncryptionOpSupported(EncryptOperationType::TLOG_ENCRYPTION);
-	CODE_PROBE(enableEncryptionForTxnStateStore, "Enable encryption for txnStateStore");
+	// If it's the first recovery the encrypt mode is not yet avilable so create the txn state store with encryption
+	// disabled. This is fine since we will not write any data to disk using this txn store.
+	state EncryptionAtRestMode encryptModeForTxnStateStore = EncryptionAtRestMode::DISABLED;
+	if (self->controllerData && self->controllerData->encryptionAtRestMode.isValid() &&
+	    self->controllerData->encryptionAtRestMode.getFuture().isValid() &&
+	    self->controllerData->encryptionAtRestMode.getFuture().isReady()) {
+		EncryptionAtRestMode encryptMode = wait(self->controllerData->encryptionAtRestMode.getFuture());
+		encryptModeForTxnStateStore = encryptMode;
+	}
```

Comment:
Can you remind me why on the first recovery we cannot wait for the encryptionAtRestMode being fulfilled?

Reply:
The `encryptionAtRestMode` promise gets set after the txnstate store is initalized. If we try wait for this then we end up in a deadlock scenario